hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanActionExecutor.java

```diff
@@ -179,6 +179,7 @@ protected Option<HoodieCleanerPlan> requestClean(String startCleanTime) {
       final HoodieInstant cleanInstant = new HoodieInstant(HoodieInstant.State.REQUESTED, HoodieTimeline.CLEAN_ACTION, startCleanTime);
       // Save to both aux and timeline folder
       try {
+        table.validateForLatestTimestamp(cleanInstant.getTimestamp());
```

Comment:
IMO, the cleaning table service does not need this check, because the lastRetained instant is kind of continuous now, even if there is a commit with geater instant time that the cleaning, the service can still take care of it.

Imagine the cleaning instant is t10, the bad case is when there is a commit with smaller instant time t9, but it finished with greater time t13, the cleaning plan is generated on t11, then the t9 commit would be skipped in the cleaning plan, but because t10 is greater than t9, it will not check the t9 anymore in incremental cleaning mode.

But, we can not fix this case even if we add this check without a lock for the whole cleaning plan generation.

Reply:
can you take a look at the scenario listed in the jira https://issues.apache.org/jira/browse/HUDI-7507 
earliest commit to retain might miss to account for some commits in some rare scenarios. So, clean also might need to have this validation

Reply:
adding retries is a good idea. 
But we could do that for all callers then? atleast whereever possible.

Reply:
Independent of the clean specific scenario, we also need to add timestamp validation to avoid the MDT compaction scenario in https://issues.apache.org/jira/browse/HUDI-7507 that you and I discussed @nsivabalan . This is since clean on the base table also performs a deltacommit on MDT. So we should keep this check.

But going back to the incremental-cleaning specific scenario
> Imagine the cleaning instant is t10, the bad case is when there is a commit with smaller instant time t9, but it finished with greater time t13, the cleaning plan is generated on t11, then the t9 commit would be skipped in the cleaning plan, but because t10 is greater than t9, it will not check the t9 anymore in incremental cleaning mode.
Yes if I understood correctly this is same scenario https://issues.apache.org/jira/browse/HUDI-7507 . Though one thing  is that afaik clean takes a table lock while computing the plan and writing it to clean.requested . So my understanding is that having clean scheduling do the timestamp validation at the beginning (before working on the plan) will maintain correctness (since any writer starting later self-abort if it generates a lower timestamp due to having to take lock as part of timestamp validation) and will "fail fast" (since if another writer created a greater instant plan before clean took the table lock clean will self-abort before doing any clean planning computation).